BOUNTY.md

```diff
@@ -0,0 +1,116 @@
+# GraphQL.NET Bounty Program
+
+The GraphQL.NET team has worked to create a program where community developers can work on specific new features
+of GraphQL.NET for compensation ("projects"). These projects are determined by GraphQL.NET maintainers to have the
+potential for high-impact improvement to the product. Design proposals that would fulfill a bounty may
+be made either by individuals or teams of developers, but the final payout will be made to only one stakeholder.
+Routine bug fixes and similar pull requests do not fall into this program at this time.
```

Comment:
```suggestion
Routine bug fixes and similar pull requests like fixing typos or outdated pieces of documentation do not fall into this program at this time.
```

Though I believe that fixing [confirmed bugs](https://github.com/graphql-dotnet/graphql-dotnet/issues?q=is%3Aopen+is%3Aissue+label%3Abug) should be paid as well.

Reply:
It's not a bad idea.  We would need to determine a structure for how this works in practice.  Having a standing price for bug fixes may be to little or too much depending on the issue.

I do like that it would encourage users of the library to fix bugs themselves rather than waiting for us to do so.

However, we would need to address how this works for us maintainers.  While I think the bounty program as written works with us maintainers (as we all agree on the bounties beforehand), bugs that you and I work on are usually spur-of-the-moment issues.  Sometimes it's just one line of code.  Sometimes it's a complex problem.  Do we just set a price and go with it per-bug?  What if the bug has multiple facets that need to be addressed?

Another option is that bug fixes are available for contributors but not maintainers.

I have not thought about it much, but I'd say in order to encourage community bug fixes, I would suggest a higher bounty price for community fixes, versus a lower price for us maintainers.  It is also easier for us to fix bugs than them, as we have greater knowledge of the codebase, so from one perspective, it's fair.

Reply:
If we do add a 'bug bounty', we should state that fixes must include a test for the bug.

Reply:
> However, we would need to address how this works for us maintainers. 

For me it does not work at all :) ... _You are a resident of any countries under U.S. sanctions_ ...

> Another option is that bug fix bounties are available for contributors but not maintainers.

I'm fine with this option.

> If we do add a 'bug bounty', we should state that fixes must include a test for the bug.

Good. Agree.

Reply:
#3430